python-package/lightgbm/basic.py

```diff
@@ -1111,13 +1188,156 @@ def __init__(self, data, label=None, reference=None,
         self.feature_penalty = None
         self.monotone_constraints = None
         self.version = 0
+        self._start_row = 0  # Used when pushing rows one by one.
 
     def __del__(self):
         try:
             self._free_handle()
         except AttributeError:
             pass
 
+    def create_sample_indices(self, total_nrow: int) -> np.ndarray:
+        """Get an array of randomly chosen indices from this ``Dataset``.
+
+        Indices are sampled without replacement.
+
+        Parameters
+        ----------
+        total_nrow : int
+            Total number of rows to sample from.
+            If this value is greater than the value of parameter ``bin_construct_sample_cnt``, only ``bin_construct_sample_cnt`` indices will be used.
+            If Dataset has multiple input data, this should be the sum of rows of every file.
+
+        Returns
+        -------
+        indices : numpy array
+            Indices for sampled data.
+        """
+        param_str = param_dict_to_str(self.params)
```

Comment:
This is filtered version of all params. Also, benefits from checking param aliases.
https://github.com/microsoft/LightGBM/blob/4167649cd50035b8237d121f706dc9f1e1b79ed0/python-package/lightgbm/basic.py#L1121-L1128

```suggestion
        param_str = param_dict_to_str(self.get_params())
```